prisma/new/schema.prisma

```diff
@@ -0,0 +1,823 @@
+generator client {
+  provider = "prisma-client-js"
+}
+
+datasource db {
+  provider = "postgresql"
+  url      = env("POSTGRES_URL")
+}
+
+/// 課程檔
+model Course {
+  id         String  @id @default(uuid()) @db.Uuid
+  /// 流水號, 可能無值
+  serial     String? @db.VarChar(5)
+  /// 課號
+  // 範例值: Common1011	
+  // 可從原 departmentAbbreviation (dpt_abbr) + codeSerial(cou_teacno) 組合來\
+  identifier String  @db.VarChar(10)
+  /// 課程識別碼
+  code       String  @db.VarChar(10)
+  /// 班次
+  // 舊db無值時會存空字串，要轉為null
+  class      String? @db.VarChar(3)
+
+  /// 全學年制, 舊系統為1 為全學年制，2為半學年制 
+  // 原forth
+  isFullYear                   Boolean
+  /// 學分
+  credits                      Decimal
+  /// 狀態
+  // 原co_chg  
+  status                       CourseStatus   @default(Normal)
+  /// 英語授課，來自舊系統NOL table english
+  isEnglishTaught              Boolean
+  /// 可充當外文領域
+  // 舊系統需join另一張table asforcou
+  asForeign                    Boolean
+  /// 為共同選修課
+  // 舊系統需join另一張table commopt
+  isCommon                     Boolean
+  /// 密集課程的週次
+  intensiveWeeks               Int[]          @default([])
+  /// 課程大綱網址
+  // 舊系統 NOL table `ntu_course`
+  // 通常是外連到師大或台科大系統課綱
+  externalSyllabusUrl          String?        @db.VarChar(200)
+  /// 修課總人數
+  // 原tno
+  totalStudentQuota            Int
+  /// 外系人數 Int
+  // 原eno
+  otherDepartmentsStudentQuota Int
+  /// 外校人數 Int
+  // 原ono
+  nonNtuStudentQuota           Int
+  /// 可加選人數 Int
+  // 原sno
+  enrollableStudentQuota       Int
+  /// 異動項目
+  // chgitem
+  changeItems                  CourseChange[]
+
+  // 1-1
+  syllabus Syllabus?
+
+  // 1-n
+  schedules                     Schedule[]
+  practiceGroups                PracticeGroup[]
+  /// 修課限制
+  limits                        Limit[]
+  /// 授課對象相關設定
+  // 一門課根據系所進行不同設定 
+  // 可以設定成不同的建議授課年級、選必修、通識, etc.
+  targets                       CourseTarget[]
+  translations                  CourseTranslation[]
+  preseletectedCourseTableItems PreseletectedCourseTableItem[]
```

Comment:
```suggestion
  preselectedCourseTableItems PreselectedCourseTableItem[]
```

typo. it's `select` instead of `seletect`.
Please also check and correct other related namings.